src/App.tsx

```diff
@@ -112,44 +193,57 @@ export const App: React.FC = () => {
         </section>
 
         {/* Hide the footer if there are no todos */}
-        <footer className="todoapp__footer">
-          <span className="todo-count">
-            3 items left
-          </span>
-
-          {/* Active filter should have a 'selected' class */}
-          <nav className="filter">
-            <a href="#/" className="filter__link selected">
-              All
-            </a>
-
-            <a href="#/active" className="filter__link">
-              Active
-            </a>
-
-            <a href="#/completed" className="filter__link">
-              Completed
-            </a>
-          </nav>
-
-          {/* don't show this button if there are no completed todos */}
-          <button type="button" className="todoapp__clear-completed">
-            Clear completed
-          </button>
-        </footer>
+        {remainingTodos !== 0 && (
+          <footer className="todoapp__footer">
+            <span className="todo-count">
+              3 items left
+            </span>
+
+            {/* Active filter should have a 'selected' class */}
+            <nav className="filter">
+              <a
+                href="#/"
+                className={classNames(
+                  'filter__link', { selected: filter === 'all' },
```

Comment:
move all filters variants into an enum and use it instead of a lot of duplicated strings (all, completed...)